### PR TITLE
Feature/ease build

### PR DESCRIPTION
**Pull request by shoshijak**
- Make life a little easier for users building CP2K with CUDA support using the toolchain script
  - Add option to specify for which GPU architecture CP2K should be compiled 
  - Make handling of NVCC flags and paths to CUDA libraries and headers more robust 
- Improve READMEs
  - Fix some typos, add links, and complete some details in READMEs
  - Make some READMEs prettier with Mardown

### CODE REVIEW

File: tools/toolchain/install_cp2k_toolchain.sh [hunk 1]
```diff
@@ -1069,8 +1149,8 @@ gen_arch_file() {
 #
 CXX         = CC
 CXXFLAGS    = \${CXXFLAGS} -I\\\${CUDA_PATH}/include -std=c++11
-GPUVER      = K20X
-NVCC        = \${NVCC} -D__GNUC__=4 -D__GNUC_MINOR__=9 -Xcompiler=--std=gnu++98
+GPUVER      = \${GPUVER}
+NVCC        = \${NVCC} -D__GNUC__=4 -D__GNUC_MINOR__=9
```
Comment (dev-zero) [inline tools/toolchain/install_cp2k_toolchain.sh:1153]: please note: for older compilers you have to pass `-std=c++11' to NVCC as well, see https://github.com/cp2k/dbcsr/commit/3069161ac6d40bb2f0728f5e97464d1ca0cc669b

Comment (shoshijak) [reply tools/toolchain/install_cp2k_toolchain.sh:1153]: I pass `--std=c++11` to NVCC via the variable `NVFLAGS`(cf line 1097 of the same file)

Comment (dev-zero) [reply tools/toolchain/install_cp2k_toolchain.sh:1153]: sorry, didn't check further down

Comment (shoshijak) [reply tools/toolchain/install_cp2k_toolchain.sh:1153]: no problem :) sorry for the late reply

**Pull request merged**